### PR TITLE
feat(Airbnb): new presence

### LINKED ISSUE #6300: Airbnb presence request
### Website name

Air BnB

### Website URL

https://www.airbnb.co.uk/

### Website logo

https://static.dezeen.com/uploads/2014/07/Airbnb-rebrand-by-DesignStudio_dezeen_468_8.jpg

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

A website about buying/renting hotel locations and reserved dates.
Preferably displaying location, logo booking dates and time on page.
Your mind is your will

### PR DESCRIPTION
**Pull request by realtofuine**
## Description 
add Airbnb Presence

closes #6300 

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

image

</details>

### CODE REVIEW

File: websites/A/Airbnb/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		"name": "Tofuine",
+		"id": "993151156652093480"
+	},
+	"service": "Airbnb",
+	"description": {
+		"en": "Airbnb, Inc. is an American company that operates an online marketplace for lodging, primarily homestays for vacation rentals, and tourism activities."
+	},
+	"url": "www.airbnb.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/UnBuyQM.png",
```
Comment (EncryptedDev) [inline websites/A/Airbnb/dist/metadata.json:13]: This is absolutely huge, could you get something smaller?

File: websites/A/Airbnb/presence.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		clientId: "993151156652093480",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+	};
+
+	presenceData.startTimestamp = browsingTimestamp;
```
Comment (EncryptedDev) [inline websites/A/Airbnb/presence.ts:11]: Assign the value within the PresenceDate object above

File: websites/A/Airbnb/presence.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	if (document.location.pathname.includes("/rooms")) {
+		presenceData.details = "Viewing rooms";
+		const room = document.querySelector(
+			"#site-content > div > div:nth-child(1) > div:nth-child(1) > div:nth-child(1) > div > div > div > div > section > div._b8stb0 > span > h1"
+		).textContent;
+		presenceData.state = room;
```
Comment (EncryptedDev) [inline websites/A/Airbnb/presence.ts:17]:
```suggestion
		presenceData.state = document.querySelector(
			"#site-content > div > div:nth-child(1) > div:nth-child(1) > div:nth-child(1) > div > div > div > div > section > div._b8stb0 > span > h1"
		).textContent;
```

File: websites/A/Airbnb/presence.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		presenceData.details = `Booking ${
+			document.querySelector("#LISTING_CARD-title").textContent
+		}`;
+		const dates = document.querySelector(
+				'[data-plugin-in-point-id="DATE_PICKER"] > div > div > div._b7b6bk > div._1qyi2pa > div._jbk4n3'
+			).textContent,
+			guests = document.querySelector(
+				'[data-plugin-in-point-id="GUEST_PICKER"] > div > div > div._b7b6bk > div._1qyi2pa > div._jbk4n3'
+			).textContent;
+		presenceData.state = `From ${dates} for ${guests}`;
```
Comment (EncryptedDev) [inline websites/A/Airbnb/presence.ts:31]: Apply the same logic here too

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Sorry for the inactivity on my behalf, managed to miss this PR 🙏🏼

File: websites/A/Airbnb/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+		"name": "Tofuine",
+		"id": "734631480541708291"
+	},
+	"service": "Airbnb",
+	"description": {
+		"en": "Airbnb, Inc. is an American company that operates an online marketplace for lodging, primarily homestays for vacation rentals, and tourism activities."
+	},
+	"url": "www.airbnb.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/DyWMYDn.png",
```
Comment (EncryptedDev) [inline websites/A/Airbnb/dist/metadata.json:13]: Must be 512x512

File: websites/A/Airbnb/presence.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+		startTimestamp: browsingTimestamp,
+	};
+
+	if (document.location.pathname.includes("/rooms")) {
+		presenceData.details = "Viewing rooms";
```
Comment (EncryptedDev) [inline websites/A/Airbnb/presence.ts:13]: This suggests that the user is viewing multiple rooms, whereas the button contradicts this and implies that the user is viewing a singular room.

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Hello 👋

There isn't a lot of depth to this presence, but it still has a strong possibility in being merged. Please try to include some more features - I have included a suggest below which may help: 
- Add a "privacy" setting to hide personal details (place names, location etc)

File: websites/A/Airbnb/presence.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		privacy = await presence.getSetting("privacy");
+
+	if (document.location.pathname.includes("/rooms")) {
+		if (privacy) {
+			presenceData.details = "Viewing a room";
+			delete presenceData.state;
+			delete presenceData.buttons;
```
Comment (EncryptedDev) [inline websites/A/Airbnb/presence.ts:17]: Neither state or buttons are set before this point so this is not needed.

File: websites/A/Airbnb/presence.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		privacy = await presence.getSetting("privacy");
+
```
Comment (EncryptedDev) [inline websites/A/Airbnb/presence.ts:12]: `document.location.pathname` is repeated many times so you should deconstruct it and reference `pathname instead`

`{ pathname } = document.location`

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**